[dhcrelay] Add debops.dhcrelay role

imrejonk opened the pull request:
`debops.dhcrelay` manages the ISC DHCP Relay Agent on Debian Stable for
forwarding of DHCPv4 messages between networks. The init script for
the isc-dhcp-relay package in Buster (and Bullseye, for that matter)
currently does not support DHCPv6, so this role currently only supports
managing dhcrelay for DHCPv4 relaying. A workaround for this can be
implemented in DebOps if there is demand for it.

This role replaces the dhcrelay functionality in `debops.dhcpd`. The
dhcrelay functionality in `debops.dhcpd` will be removed in the next
role update.

imrejonk committed:
[dhcrelay] Add debops.dhcrelay role

`debops.dhcrelay` manages the ISC DHCP Relay Agent on Debian Stable for
forwarding of DHCPv4 messages between networks. The init script for
the isc-dhcp-relay package in Buster (and Bullseye, for that matter)
currently does not support DHCPv6, so this role currently only supports
managing dhcrelay for DHCPv4 relaying. A workaround for this can be
implemented in DebOps if there is demand for it.

This role replaces the dhcrelay functionality in `debops.dhcpd`. The
dhcrelay functionality in `debops.dhcpd` will be removed in the next
role update.

tasosalvas reviewed (changes requested):
Missing [upgrade](https://github.com/debops/debops/blob/master/docs/news/upgrades.rst) instructions for people using the feature via `debops.dhcpd`.

imrejonk commented:
@tasosalvas Good point, I added the upgrade notes.

tasosalvas reviewed (dismissed):
Looks great, thanks!

imrejonk commented:
And I also added the JANE test for `debops.dhcrelay` to .gitlab-ci.yml. Already tested successfully against our GitLab runners.

drybjed commented on ansible/roles/dhcrelay/handlers/main.yml:
```diff
@@ -0,0 +1,9 @@
+- name: Restart isc-dhcp-relay
+  service:
+    name: 'isc-dhcp-relay'
+    state: 'restarted'
```
The handlers should now be added in the `global_handlers` role, in a separate file. Don't forget to import that role in the task list.

drybjed commented on ansible/roles/dhcrelay/tasks/main.yml:
```diff
@@ -0,0 +1,40 @@
+           --divert /etc/default/isc-dhcp-relay.dpkg-divert
+           --rename /etc/default/isc-dhcp-relay
+  args:
+    creates: '/etc/default/isc-dhcp-relay.dpkg-divert'
```
Perhaps you could convert this task to use the `dpkg_divert` Ansible module?

drybjed commented on ansible/roles/dhcrelay/templates/etc/ansible/facts.d/dhcrelay.fact.j2:
```diff
@@ -0,0 +1,28 @@
+
+if output['installed']:
+    output['version'] = subprocess.check_output(
+        ['dhcrelay', '--version'], stderr=subprocess.STDOUT
```
Ah, so that's the source of the role name... Sounds good then. I would go with `dhcp_relay`, but if the command line name exists, that's fine with me. :+1:

imrejonk replied:
Yeah, naming things is one of the two hard things in IT, next to cache invalidation and off-by-one errors :)
I did indeed name this role dhcrelay after the command line name, just as with dhcpd. It's also a bit shorter than dhcp_relay and maybe less ambiguous (there could be more software out there that does DHCP relaying).

drybjed reviewed (changes requested):
Nice work, especially the documentation. :-) I know that you lifted the tasks and handlers from the `dhcpd` role, but perhaps you could use the occasion to modernize the code a bit. :-) Check the comments for suggestions.

imrejonk commented:
@drybjed Thanks for the feedback! All your comments should be addressed now. I also added Galaxy meta and Ansible collection information and updated the JANE_LOG_PATTERN.

imrejonk commented:
Whoops, forgot to add `- import_tasks: 'dhcrelay.yml'` to `global_handlers/handlers/main.yml`...
Fixed now.

drybjed reviewed (approved):
Nicely done, thanks!
